test/api.spec.js

```diff
@@ -0,0 +1,482 @@
+const supertest = require('supertest');
+const app = require('../server/index.js');
+const request = supertest.agent(app);
+
+const db = require('./../database');
+
+describe('Test GET /books/:id', () => {
+
+  let mock;
+  afterEach(() => {
+    if(mock)
+      mock.mockRestore();
+  });
+
+  test('Should return a 200 for valid bookId', (done) => {
+    mock = jest.spyOn(db, 'getBookInfo');
+
+    const rows = [{ id: 1 }];
```

Comment:
It's unclear to me what you're trying to do here with the mockImplementation function. Could you please explain a little? 
Are you mocking the response to getBookInfo to be [{id:1}]? If so, doesn't that defeat the purpose of the test?

Reply:
It's mocking getBookInfo and testing my route implementation. Because it should test the route, not the database functionality. In the unit tests, shouldn't need access to external resources such as db, an HTTP request and so on. So all external resources are mocked.